packages/frameworks/src/types.ts

```diff
@@ -1,50 +1,174 @@
 import { Route } from '@vercel/routing-utils';
 
 export interface FrameworkDetectionItem {
+  /**
+   * A file path
+   * @example package.json
+   */
   path: string;
+  /**
+   * A matcher
+   * @example \"(dev)?(d|D)ependencies\":\\s*{[^}]*\"next\":\\s*\".+?\"[^}]*}
+   */
   matchContent?: string;
 }
 
 export interface SettingPlaceholder {
+  /**
+   * A placeholder value for when the framework has not a predefined one
+   * @example `npm run build` or `next build`
+   */
   placeholder: string;
 }
 
 export interface SettingValue {
+  /**
+   * A predefined setting for the detected framework
+   * @example next dev --port $PORT
+   */
   value: string;
 }
 
 export type Setting = SettingValue | SettingPlaceholder;
 
+/**
+ * Framework detection information.
+ */
 export interface Framework {
+  /**
+   * Name of the framework
+   * @example Next.js
+   */
   name: string;
+  /**
+   * A unique identifier for the framework
+   * @example nextjs
+   */
   slug: string | null;
+  /**
+   * A URL to the logo of the framework
+   * @example https://raw.githubusercontent.com/vercel/vercel/master/packages/frameworks/logos/next.svg
+   */
   logo: string;
+  /**
+   * A URL to a deployed example of the framework
+   * @example https://nextjs.now-examples.now.sh
+   */
   demo?: string;
+  /**
+   * A marketing tagline for the framework
+   * @example "Next.js makes you productive with React instantly — whether you want to build static or dynamic sites."
+   */
   tagline?: string;
+  /**
+   * A URL to the official website of the framework
+   * @example https://nextjs.org
+   */
   website?: string;
+  /**
+   * Short description of the framework
+   * @example A Next.js app and a Serverless Function API.
+   */
   description: string;
+  /**
+   * A ponderated value to sort matching frameworks
+   * @example 1
+   */
   sort?: number;
-  useRuntime?: { src: string; use: string };
+  /**
+   * Runtime configuration required to run the framework in Vercel
+   */
+  useRuntime?: {
+    /**
+     * Runtime source
+     * @example "package.json"
+     */
+    src: string;
+    /**
+     * Runtime
+     * @example @vercel/next
```

Comment:
```suggestion
     * @example "@vercel/next"
```

Reply:
I'm gonna leave it for now to be consistent with the others